core/src/components/select/select-interface.ts

```diff
@@ -2,6 +2,6 @@ export type SelectInterface = 'action-sheet' | 'popover' | 'alert';
 
 export type SelectCompareFn = (currentValue: any, compareValue: any) => boolean;
 
-export interface SelectChangeEventDetail {
-  value: any | any[] | undefined | null;
+export interface SelectChangeEventDetail<T = any> {
+  value: T;
```

Comment:
Both `undefined` and `null` and `any[]` are sub-types of `any`, so this is still fully compatible. But if there's a case where `value` doesn't exist at all in `event.detail`, it should probably be

```suggestion
  value?: T;
```

I'm not aware that this could happen though.

Reply:
I believe `value` is always present, so we should be good.